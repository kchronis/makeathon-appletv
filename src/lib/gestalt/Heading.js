/**
 * WIP - Gestalt like Heading
 * https://github.com/pinterest/gestalt/tree/master/src/Heading
 * @flow
 */

import * as React from 'react';
import { View, StyleSheet, Text as ReactText } from 'react-native';
import type { StyleObj } from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import PIColor, { gestaltColorToColor } from '../styles/picolor';
import PIText from '../styles/pitext';
import { isSmallPhoneDevice } from '../Utils';

const headingSizeToPITextSize = {
  xs: PIText.displayExtraSmallFontSize,
  sm: PIText.displaySmallFontSize,
  md: PIText.displayMediumFontSize,
  lg: PIText.displayLargeFontSize,
  xl: PIText.displayExtraLargeFontSize,
};

const headingSmallPhoneSizeToPITextSize = {
  xs: PIText.displaySmallPhoneExtraSmallFontSize,
  sm: PIText.displaySmallPhoneSmallFontSize,
  md: PIText.displaySmallPhoneMediumFontSize,
  lg: PIText.displaySmallPhoneLargeFontSize,
  xl: PIText.displaySmallPhoneExtraLargeFontSize,
};

const getHeadingSizeToPITextSize = (textSize: string) =>
  isSmallPhoneDevice
    ? headingSmallPhoneSizeToPITextSize[textSize]
    : headingSizeToPITextSize[textSize];

type Props = {|
  // accessibilityLevel?: 1 | 2 | 3 | 4 | 5 | 6, // Not available in RN
  children?: any,
  color?:
    | 'primaryBackground'
    | 'darkGray'
    | 'lightGray'
    | 'superLightGray'
    | 'red'
    | 'elephantPinBackground'
    | 'white',
  // id?: string, // Not available in RN
  size: 'xs' | 'sm' | 'md' | 'lg' | 'xl',
  truncate?: boolean,

  style?: StyleObj, // React Native only: Should go away eventually
|};

export default function Heading(props: Props) {
  const {
    // accessibilityLevel,
    children,
    color = 'darkGray',
    // id = null,
    size,
    truncate = false,
    style,
  } = props;

  const headingStyles = {
    color: gestaltColorToColor[color],
    fontSize: getHeadingSizeToPITextSize(size),
    fontFamily: PIText.defaultFont,
    fontWeight: 'bold',
    paddingBottom: 0,
    paddingTop: 0,
    textAlign: 'center',
  };

  return (
    <ReactText
      style={StyleSheet.flatten([headingStyles, style])}
      numberOfLines={truncate ? 1 : 0}
    >
      {children}
    </ReactText>
  );
}

// Playground support

// eslint-disable-next-line import/first
import { type PlaygroundDefineType } from '../../screens/Playground/Playground';

module.exports.__cards__ = (define: PlaygroundDefineType) => {
  define('Default Heading', () => (
    <View
      style={{
        backgroundColor: PIColor.primaryBackgroundColor,
        paddingVertical: 5,
        paddingHorizontal: 10,
      }}
    >
      <Heading size={'md'} style={{ textAlign: 'center' }}>
        Medium heading in default color
      </Heading>
    </View>
  ));
  define('Small Truncated Heading', () => (
    <View
      style={{
        backgroundColor: PIColor.primaryBackgroundColor,
        paddingVertical: 5,
        paddingHorizontal: 10,
      }}
    >
      <Heading size={'sm'} truncate>
        Small truncated heading text text text text
      </Heading>
    </View>
  ));
};
