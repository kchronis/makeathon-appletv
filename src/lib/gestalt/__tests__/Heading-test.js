/* eslint-env jest */
import React from 'react';
import { create } from 'react-test-renderer';
import Heading from '../Heading';

test('Heading small', () => {
  const tree = create(<Heading size="sm" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Heading medium with text', () => {
  const tree = create(<Heading size="md">Heading</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});
