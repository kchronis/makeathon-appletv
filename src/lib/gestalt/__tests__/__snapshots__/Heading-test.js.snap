// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading medium with text 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    Object {
      "color": "#555",
      "fontFamily": "HelveticaNeue",
      "fontSize": 36,
      "fontWeight": "bold",
      "paddingBottom": 0,
      "paddingTop": 0,
      "textAlign": "center",
    }
  }
>
  Heading
</Text>
`;

exports[`Heading small 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={0}
  style={
    Object {
      "color": "#555",
      "fontFamily": "HelveticaNeue",
      "fontSize": 24,
      "fontWeight": "bold",
      "paddingBottom": 0,
      "paddingTop": 0,
      "textAlign": "center",
    }
  }
/>
`;
